flink/v1.17/flink/src/main/java/org/apache/iceberg/flink/sink/RowDataTaskWriterFactory.java

```diff
@@ -108,7 +130,10 @@ public RowDataTaskWriterFactory(
   @Override
   public void initialize(int taskId, int attemptId) {
     this.outputFileFactory =
-        OutputFileFactory.builderFor(table, taskId, attemptId).format(format).build();
+        OutputFileFactory.builderFor(initTable, taskId, attemptId)
+            .format(format)
+            .ioSupplier(() -> tableSupplier.get().io())
```

Comment:
Could it be that the table io is changing during the lifetime of the table? Do we plan to accommodate those changes?

Reply:
Using the current table's FileIO to perform file operations should be safe, I think it actually desirable. For example, if write properties on the table are updated, those will be applied at runtime rather than requiring a restart (only if table reload is explicitly enabled).

Reply:
This was updated, so the table reload is explicitly called now, rather than done as part of `get()`

Reply:
Thanks!